options/internal/gcc/stack_protector.cpp

```diff
@@ -0,0 +1,28 @@
+#include <stdint.h>
+#include <string.h>
+#include <mlibc/debug.hpp>
+#include <mlibc/stack_protector.hpp>
+#include <mlibc/thread.hpp>
+#include <bits/ensure.h>
+
+uintptr_t __stack_chk_guard = 0;
```

Comment:
This belong to an external ABI, right? What's its purpose?

Reply:
Some platforms/toolchains will emit a reference to ``__stack_chk_guard`` for the stack canary, others will get it from ``fs:0x28``.

Reply:
Alright. How does `__stack_chk_guard` work in multi-threaded programs?